src/features/adr/pages/PatientForm.tsx

```diff
@@ -33,19 +34,78 @@ const useStyles = stylesFactory({
 export interface PatientFormProps {
   onSubmit: (data: any) => void;
   step: number;
+  jsonSchema: any;
+  patientName?: string;
+  setJsonSchema: any;
 }
 
-export const PatientForm: FC<PatientFormProps> = ({ onSubmit, step }) => {
+const mapHistory = (history: PatientHistory[]) =>
+  history
+    .filter((h) => h.transLines.some((t) => t.itemLine.item.is_vaccine))
+    .map((h, index) => {
+      const transLine = h.transLines.length
+        ? h.transLines[0]
+        : ({} as Transaction);
+      const { medicineAdministrator, item_name = "" } = transLine;
+      const vaccinator = medicineAdministrator
+        ? `  Vaccinator: ${medicineAdministrator.first_name} ${medicineAdministrator.last_name}`
+        : "";
+
+      return `${index + 1}. ${item_name}  Date: ${h.confirm_date}${vaccinator}`;
+    });
+
+export const PatientForm: FC<PatientFormProps> = ({
+  onSubmit,
+  step,
+  jsonSchema,
+  patientName,
+  setJsonSchema,
+}) => {
   const classes = useStyles();
   const { data, setData } = useStep(step);
   const { isGuest } = useAuth();
   const [canContinue, setCanContinue] = useState(false);
   const [manualEntry, setManualEntry] = useState(isGuest);
+  const [patient, setPatient] = useState(
+    data?.patient || { first: "", last: "", date_of_birth: null }
+  );
   const { messages } = useTranslations();
+  const {
+    loading: historyLoading,
+    data: history,
+    searched,
+    searchOnline: historySearch,
+  } = usePatientHistory();
 
   const onNextHook = () => canContinue;
-  const patient = data?.patient || { first: "", last: "", date_of_birth: null };
-  const setPatient = (patient: Patient) => setData({ patient });
+  const setPatientData = (patient: Patient) => {
+    if (data?.patient?.ID === patient.ID) return;
+    setData({ patient });
+    setPatient(patient);
+    patientName = `${patient.first} ${patient.last}`;
+    if (!historyLoading) historySearch(patient.ID);
+  };
+
+  useEffect(() => {
+    if (!jsonSchema) return;
+
+    const { properties = {} } = jsonSchema;
+    const { causes = {} } = properties;
+    const { items = {} } = causes;
+
+    if (!items) {
+      setJsonSchema(jsonSchema);
+      return;
+    }
+
+    items.enum = history.length
+      ? mapHistory(history)
+      : ["No vaccination history"];
```

Comment:
Was wondering if using `enumNames` and the `ID` of the actual transaction would make it easier to find the transaction for an adr at some later stage?

Also wondering if we need to use localization anywhere here?

Reply:
Yes, I thought about enumNames.. I like that idea a lot.
fb522ef

have added translations too!